src/analyzer.ts

```diff
@@ -434,39 +415,89 @@ export class Analyzer {
         },
       });
     }
+    // It's important to in-place update the placeholder object, instead of
+    // creating a new object, because other configs may be referencing this
+    // exact object in their dependencies.
+    const remainingConfig: LocallyValidScriptConfig = {
+      ...placeholder,
```

Comment:
Why do you need this spread if you just assign back into `placeholder`?

Reply:
So that the `:LocallyVaildScriptConfig` type annotation can be there, so that we know we're ending up with an object that fully satisfies that interface.